.github/actions/optimize-paths-filter/action.yml

```diff
@@ -0,0 +1,39 @@
+---
+name: Common paths filter for optimize
+
+description: Common filters to detect and group changes against a base branch
+
+outputs:
+  backend-changes:
+    description: Output whether any backend has changed
+    value: >-
+      ${{
+        github.event_name == 'push' ||
+        steps.filter-common.outputs.backend-change == 'true'
+      }}
+  frontend-changes:
+    description: Output whether any frontend has changed
+    value: >-
+      ${{
+        github.event_name == 'push' ||
+        steps.filter-common.outputs.frontend-change == 'true'
+      }}
+runs:
+  using: composite
+  steps:
+    - uses: dorny/paths-filter@v3
+      id: filter-common
+      with:
+        base: ${{ github.event.merge_group.base_ref || '' }}
+        ref: ${{ github.event.merge_group.head_ref || github.ref }}
+        filters: |
+          backend-change:
+            - optimize/backend/**
+            - optimize/upgrade/**
+            - optimize/util/**
+            - optimize/plugins/**
```

Comment:
I _think_ we are deleting this whole module "soon" as it is not C8. Does it make sense to already exclude it here?

Reply:
I think then we should leave it as it is and she can clean this file up as well in scope of her changes